lldb/source/Plugins/TypeSystem/Swift/TypeSystemSwiftTypeRef.h

```diff
@@ -505,6 +506,8 @@ class TypeSystemSwiftTypeRef : public TypeSystemSwift {
 
   /// All lldb::Type pointers produced by DWARFASTParser Swift go here.
   ThreadSafeDenseMap<const char *, lldb::TypeSP> m_swift_type_map;
+  swift_demangle::LRUCache<std::string, CompilerType> m_canonical_types_cache;
```

Comment:
We'll probably need to make sure these data structures are thread safe.

Reply:
Yes, good point. I thought about this but then forgot. I'll add a mutex